mgmtfn/k8splugin/driver.go

```diff
@@ -0,0 +1,370 @@
+		TenantName:  req.Tenant,
+		NetworkName: req.Network,
+		ServiceName: req.Group,
+		EndpointID:  req.EndpointID,
+	}
+
+	var delResp master.DeleteEndpointResponse
+	err2 := cluster.MasterPostReq("/plugin/deleteEndpoint", &delReq, &delResp)
+
+	if err1 != nil {
+		return err1
+	}
+
+	return err2
+}
+
+// createEP creates the specified EP in contiv
+func createEP(req *epSpec) (*epAttr, error) {
+
+	// if the ep already exists, treat as error for now.
+	netID := req.Network + "." + req.Tenant
+	ep, err := netdGetEndpoint(netID + "-" + req.EndpointID)
+	if err == nil {
+		return nil, fmt.Errorf("EP %s already exists", req.EndpointID)
+	}
+
+	// Build endpoint request
+	mreq := master.CreateEndpointRequest{
+		TenantName:  req.Tenant,
+		NetworkName: req.Network,
+		ServiceName: req.Group,
+		EndpointID:  req.EndpointID,
+		ConfigEP: intent.ConfigEP{
+			Container:   req.EndpointID,
+			Host:        pluginHost,
+			ServiceName: req.Group,
+		},
+	}
+
+	var mresp master.CreateEndpointResponse
+	err = cluster.MasterPostReq("/plugin/createEndpoint", &mreq, &mresp)
+	if err != nil {
+		epCleanUp(req)
+		return nil, err
+	}
+
+	log.Infof("Got endpoint create resp from master: %+v", mresp)
+
+	// Ask netplugin to create the endpoint
+	err = netPlugin.CreateEndpoint(netID + "-" + req.EndpointID)
+	if err != nil {
+		log.Errorf("Endpoint creation failed. Error: %s", err)
+		epCleanUp(req)
+		return nil, err
+	}
+
+	ep, err = netdGetEndpoint(netID + "-" + req.EndpointID)
+	if err != nil {
+		epCleanUp(req)
+		return nil, err
+	}
+
+	log.Debug(ep)
+	// need to get the subnetlen from nw state.
+	nw, err := netdGetNetwork(netID)
+	if err != nil {
+		return nil, err
+	}
+
+	epResponse := epAttr{}
+	epResponse.PortName = ep.PortName
+	epResponse.IPAddress = ep.IPAddress + "/" + strconv.Itoa(int(nw.SubnetLen))
+
+	return &epResponse, nil
+}
+
+// getLink is a wrapper that fetches the netlink corresponding to the ifname
+func getLink(ifname string) (netlink.Link, error) {
+	// find the link
+	link, err := netlink.LinkByName(ifname)
+	if err != nil {
+		if !strings.Contains(err.Error(), "Link not found") {
+			log.Errorf("unable to find link %q. Error: %q", ifname, err)
+			return link, err
+		}
+		// try once more as sometimes (somehow) link creation is taking
+		// sometime, causing link not found error
+		time.Sleep(1 * time.Second)
+		link, err = netlink.LinkByName(ifname)
+		if err != nil {
+			log.Errorf("unable to find link %q. Error %q", ifname, err)
+		}
+		return link, err
+	}
+	return link, err
+}
+
+// nsToPID is a utility that extracts the PID from the netns
+func nsToPID(ns string) (int, error) {
+	// Make sure ns is well formed
+	ok := strings.HasPrefix(ns, "/proc/")
+	if !ok {
+		return -1, fmt.Errorf("Invalid nw name space: %v", ns)
+	}
+
+	elements := strings.Split(ns, "/")
+	return strconv.Atoi(elements[2])
+}
+
+// setIfAttrs sets the required attributes for the container interface
+func setIfAttrs(ifname, netns, cidr, newname string) error {
```

Comment:
Yes I see the netns lib. I am not sure it will do everything we need. But I can look at that separately.